app/jobs/add_workflow_job.rb

```diff
@@ -2,26 +2,29 @@
 
 ##
 # Job to add a workflow to and object
-class AddWorkflowJob < GenericJob
-  attr_reader :workflow_name
+class AddWorkflowJob < BulkActionJob
```

Comment:
Most of the boilerplate is in BulkActionJob (or BulkActionCsvJob for bulk actions that take CSV as input).

Each bulk action has to implement a subclass of BulkActionItem which performs processing for a single item.